src/brogue/Monsters.c

```diff
@@ -1562,21 +1562,21 @@ boolean monsterCanShootWebs(creature *monst) {
 // Returns approximately double the actual (quasi-euclidian) distance.
 short awarenessDistance(creature *observer, creature *target) {
     long perceivedDistance;
-
-    // start with base distance
-    if ((observer->status[STATUS_LEVITATING]
-         || (observer->info.flags & MONST_RESTRICTED_TO_LIQUID)
-         || (observer->info.flags & MONST_IMMOBILE)
-         || (observer->bookkeepingFlags & MB_SUBMERGED)
-         || ((observer->info.flags & MONST_IMMUNE_TO_WEBS) && monsterCanShootWebs(observer)))
-        && ((target == &player && (pmap[observer->xLoc][observer->yLoc].flags & IN_FIELD_OF_VIEW))
-            || (target != &player && openPathBetween(observer->xLoc, observer->yLoc, target->xLoc, target->yLoc)))) {
-            // if monster flies or is immobile or waterbound or underwater or can cross pits with webs,
-            // use absolute distance.
-            perceivedDistance = scentDistance(observer->xLoc, observer->yLoc, target->xLoc, target->yLoc);
-        } else {
-            perceivedDistance = (rogue.scentTurnNumber - scentMap[observer->xLoc][observer->yLoc]); // this value is double the apparent distance
-        }
+    
+    // When determining distance from the player for purposes of monster state changes
+    // (i.e. whether they start or stop hunting), take the scent value of the monster's tile
+    // OR, if the monster is in the player's FOV (including across chasms, through green crystal, etc.),
+    // the direct distance -- whichever is less.
+    // This means that monsters can aggro within stealth range if they're on the other side
```

Comment:
Can you elaborate more on how this interacts with stealth radius (and stealth builds, by extension)?

Reply:
Previously, only flying and aquatic/submerging monsters would aggro across a chasm, lake or obstruction crystal (although still only from within stealth radius). Other monsters used only the scent system, which does not propagate a gradient across typically impassable terrain. Now, all monsters will aggro like flying monsters. If they can see you and they're within stealth radius, they're able to notice you even if they can't get to you. If you leave their FOV, they'll try to wander to where they last saw you. All of this was and still is limited by stealth radius.

Reply:
Ok thank you.  The phrasing confused me a little.  I did just prove to myself that a high-value sneak ring still works as expected (monsters go back to wandering if they're far enough) even when obstruction blasts are around.